proposals/scripting-support.md

```diff
@@ -0,0 +1,434 @@
+
+# Kotlin Scripting support
+
+*Replaces [Script Definition Template](https://github.com/Kotlin/KEEP/blob/master/proposals/script-definition-template.md)
+proposal.*
+
+## Feedback
+
+Discussion of this proposal is held in [this issue](https://github.com/Kotlin/KEEP/issues/75)
+
+## Motivation
+
+- Define Kotlin scripting and it's applications
+- Describe intended use cases for the Kotlin scripting
+- Define scripting support that is: 
+  - applicable to the all Kotlin platforms
+  - provides sufficient control of interpretation and execution of scripts
+  - simple enough to configure and customize
+  - provides usable default components and configurations for the typical use cases    
+- Provide basic examples of the scripting usage and implementation
+- Address the issues found during the public usage of the current scripting support 
+
+## Applications
+
+* Build scripts (Gradle/Kobalt)
+* Test scripts (Spek)
+* Command-line utilities
+* Routing script for ktor (get(“/hello”) {...} on top level)
+* Type-safe configuration files (TeamCity)
+* In-process scripting and REPL for IDE
+* Consoles like IPython Notebook
+* Game scripting engines
+* ...
+
+## Basic definitions
+
+- **Script** - a text file written in Kotlin language but allowing top-level statements and having access to some 
+  implicit (not directly mentioned in the script text) properties, as if the whole script body is a body of an implicit
+  function
```

Comment:
I meant `parameters` in fact. But the statement is ambiguous since we assume some implicit imports that are not referenced in the body as well.